vnext/Microsoft.ReactNative/Utils/Helpers.cpp

```diff
@@ -119,9 +123,10 @@ bool IsWinUI3Island() {
 #endif
 }
 
-bool IsFabricEnabled(winrt::Microsoft::ReactNative::IReactPropertyBag const & /*properties*/) {
+bool IsFabricEnabled(winrt::Microsoft::ReactNative::IReactPropertyBag const &properties) {
 #ifdef USE_FABRIC
-  return false; // Once we add an instance property to control switching to fabric, this will check that.
+  return winrt::Microsoft::ReactNative::Composition::implementation::CompositionUIService::GetCompositionContext(
```

Comment:
Is it possible that we might have multiple root views in the same application, some using Fabric / Composition and some using Paper / XAML? If so, can we avoid the mistakes of the past and add a generalized `ReactUIService` and that returns something analogous to `CompositionUIService` / `XamlUIService` per root view?

Reply:
I'm hoping that will be possible, but I'm not 100% sure it will be.  Since fabric/paper are going to be so different there are some things that do not have any context as to what kind of rootview they are attached to but we need to do different things depending on fabric vs paper.

The places that this is being used do not have any kind of root view context. Currently this is being used in ImageLoader: how can we know which kind of rootview the image is being loaded for?  If the JS calls Image.prefetch/Image.getSize, should we use the XAML image decoder or the wincodec one?  That call has no root associated with it, and may not be associated with any specific root.  Maybe we have to switch everything to use wincodec all the time.  

RedBox/LogBox - These are global events.  Again, hopefully long term we can realign these to do the same thing all the time.  I'm hoping we can switch to creating a new window for RedBoxes, rather than trying to take over an existing window. -- We could run redbox in its own UI thread, thus avoiding any conflicts with the apps UI thread.

Another instance that I haven't dont in fabric yet is animations.  The animation module has a bunch of XAML dependencies currently, and when I took a quick look currently the calls to the native module to not always know their context either. This has at least a potential to be hard to get working in a way that allows both paper and fabric in the same instance. -- But I dont know yet.

Reply:
> If the JS calls Image.prefetch/Image.getSize, should we use the XAML image decoder or the wincodec one?
 
AlertModule has the same limitation - and the "right" fix is to add a root tag hint for it. I wonder if it also makes sense to add a root tag hint for prefetching images. After all, even in iOS and Android, images may be fetched for different roots on different displays that may want to use different approaches for prefetching.

Also, why can't we use the XAML image decoder for Image.getSize regardless of where it gets rendered?

Reply:
We could use it for getSize.  But we cannot use it to render the image, since we are not using a XAML element for image, but rendering directly to a composition visual.  I'd rather use the same decoder for rendering as we use for measuring, to ensure the results match.

Also at least in the near term, internally Office is going to be using fabric without loading the XAML dll.